### PR TITLE
feat: Updates GA status for Atlas Stream resources 

### PR DESCRIPTION
**Pull request by lantoli**
## Description

Updates GA status for Atlas Stream resources 

Link to any related issue(s): CLOUDP-237008

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: internal/provider/provider_sdk2.go [hunk 1]
```diff
@@ -136,10 +130,7 @@ func NewSdkV2Provider(proxyPort *int) *schema.Provider {
 		DataSourcesMap: getDataSourcesMap(),
 		ResourcesMap:   getResourcesMap(),
 	}
-	addPreviewFeatures(provider)
-
```
Comment (AgustinBettati) [inline internal/provider/provider_sdk2.go:140]: makes sense that this can be removed completely 👋

**Review by AgustinBettati — APPROVED** [review]
LGTM

File: internal/provider/provider_sdk2.go (hunk 1 above)
Comment (lantoli) [reply internal/provider/provider_sdk2.go:140]: yes, I only kept it for the new fw

**Review by oarbusi — APPROVED** [review]

**Review by EspenAlbert — APPROVED** [review]
Probably can remove from the root `README.md`: [## Preview Features](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/9d4c36c9c2a5b905924e105b4fc8b41815915c73/README.md#L30)

**Comment by lantoli** [comment]
> Probably can remove from the root README.md: [## Preview Features](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/9d4c36c9c2a5b905924e105b4fc8b41815915c73/README.md#L30)

@EspenAlbert i kept the README.md section on purpose in case we have other preview features in the future

**Review by pierwill — APPROVED** [review]

**Review by Zuhairahmed — APPROVED** [review]

**Pull request merged**